crates/nu-protocol/src/plugin_signature.rs

```diff
@@ -0,0 +1,246 @@
+use crate::{PluginExample, Signature};
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::engine::Command;
+use crate::{BlockId, Category, Flag, PositionalArg, SyntaxShape, Type};
+
+/// A simple wrapper for Signature, includes examples.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct PluginSignature {
```

Comment:
Not really sure if there is a simple way to define `PluginSignature`, it's just `Signature` with additional examples, 

But I don't think it's a good idea to add `examples: Vec<Example>` attributes into `Signature` itself, because for normal commands, examples don't belongs to Signature, so they are mostly useless in nushell core.

Reply:
If we add `examples: Vec<Example>` as Signature attribute, the diff will be small, but I think it makes us hard to maintain it in the future.

Reply:
I think for the current state of things your wrapper type approach sounds reasonable.